test_helpers/index.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * @adonisjs/view
+ *
+ * (c) AdonisJS
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+import { IgnitorFactory } from '@adonisjs/core/factories'
+
+export const BASE_URL = new URL('./tmp/', import.meta.url)
+
+export async function setupApp(
+  environment: 'web' | 'repl',
+  additionalConfig: Record<string, any> = {}
+) {
+  const IMPORTER = (filePath: string) => {
+    if (filePath.startsWith('./') || filePath.startsWith('../')) {
+      return import(new URL(filePath, BASE_URL).href)
+    }
+    return import(filePath)
+  }
+
+  const ignitor = new IgnitorFactory()
+    .merge({
+      rcFileContents: {
+        providers: ['../../providers/view_provider.js'],
+      },
+    })
+    .withCoreConfig()
+    .withCoreProviders()
+    .merge({ config: additionalConfig })
+    .create(BASE_URL, { importer: IMPORTER })
```

Comment:
It seems to me that it is, but I'm curious to know if this is now the desired approach to create and test an application

Reply:
Yup, this is the way to go. Each package exposes factories that can be used for testing. Lemme know if you think any part of these factories can be improved

Reply:
Yes this is super useful !

I would say that the only thing that seems a bit cumbersome is to add an `importer`. Maybe we could have a default one ?

Reply:
The default one shipped via package won't work for relative imports. In case of relative imports, if the `import` method is defined inside a package, then it will look for files relative to the package and not the host app.

Alternative is to use `import.meta.resolve` API. But that is experimental right now and also a little slow on performance